README.md

```diff
@@ -2,65 +2,220 @@
+        /// This is the modified handler to use the wrapper over a non-void synchronous function.
+        /// </summary>
+        public string FunctionWithReturn(string input, ILambdaContext context)
+        {
+            return s_functionWrapper.Invoke(OriginalFunctionWithReturn, input, context);
+        }
+```
+
+Check the sample project for a working example:
+
+- [Sample Lambda Functions](./src/SampleLambdaFunctions)
+
+The are overloads of the `Invoke` method with different signatures to support the typical function signatures and specific ones to support API Gateway functions:
+
+- `InvokeAPIGatewayProxyAsync`
+- `InvokeAPIGatewayHttpApiV2ProxyAsync`
+
+All of the methods above support enrichment of spans via optional parameters:
+
+- `operationName` allows a custom and more friendly name for the span, if not specified defaults to the Lambda function name.
+- `tags` lists extra key value pairs to be added to the span.
+
+## Step 2: Locate Your Organization's Realm and Access Token
 
 To locate your realm:
 
 1. Open SignalFx and in the top, right corner, click your profile icon.
 2. Click **My Profile**.
 3. Next to **Organizations**, review the listed realm.
 
-To set your realm, use a subdomain, such as ingest.us1.signalfx.com or ingest.eu0.signalfx.com. You will use the realm subdomain to set `SIGNALFX_API_HOSTNAME` variable in the next step.
+The realm will be used to configure the SignalFx endpoint that ingests your data.
+
+To obtain an access token:
+
+1. Open SignalFx and in the top, right corner, click your profile icon.
+2. Click **My Profile**.
+3. On the left panel click **Access Tokens**, select or create a new token as appropriate.
+
+The access token will be used to associate the data to your organization on SignalFx.
 
 ## Step 3: Set Lambda function environment variables
 
-1. Set authentication token:
-    ```text
-     SIGNALFX_AUTH_TOKEN=signalfx token
+1. Set `SIGNALFX_ACCESS_TOKEN` with your access token. Review the following example. 
+    ```bash
+     SIGNALFX_ACCESS_TOKEN=<access-token>
     ```
-2. Review optional parameters: 
-    ```text
-     SIGNALFX_API_HOSTNAME=[ingest.us0.signalfx.com]
-     SIGNALFX_API_PORT=[443]
-     SIGNALFX_API_SCHEME=[https]
-     SIGNALFX_SEND_TIMEOUT=milliseconds for signalfx client timeout [2000]
+
+2. Set `SIGNALFX_ENDPOINT_URL` with your organization's realm. Review the following example.
+    ```bash
+     SIGNALFX_ENDPOINT_URL=https://ingest.<realm>.signalfx.com
+    ```
+
+Note that the you'll need to specify the full path when sending to OpenTelemtry Collector i.e, `http://<otel-collector-host>:9411/api/v2/spans`.
+
+3. (Optional) Globally enable/disable metrics and/or tracing by setting SIGNALFX_TRACING_ENABLED and/or SIGNALFX_METRICS_ENABLED.
+    ```bash
+    SIGNALFX_TRACING_ENABLED=true  [defaults to true]
+    SIGNALFX_METRICS_ENABLED=false [defaults to false]
+    ```
+
+4. (Optional) Enable context propagation (currently only supports B3 propagation).
```

Comment:
when would i want to disable this?

Reply:
I actually think that if you have your service open to the Internet it can be risky to have this enabled by default. So the default is actually disabled, users need to opt-in.